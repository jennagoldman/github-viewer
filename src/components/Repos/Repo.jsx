import React from 'react';

import PropTypes from 'prop-types';

const Repo = ({ repoUrl, repoName }) => (
  <>
    <a href={repoUrl} target="_blank" rel="noopener noreferrer">{repoName}</a>
  </>
);

Repo.propTypes = {
  repoUrl: PropTypes.string.isRequired,
  repoName: PropTypes.string.isRequired
};

export default Repo;
