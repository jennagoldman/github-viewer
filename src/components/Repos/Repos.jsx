import React from 'react';
import Repo from './Repo.jsx';

import PropTypes from 'prop-types';

const Repos = ({ repos }) => {
  const repoElements = repos.map(repo => (
    <li key={repo.id}>
      <Repo {...repo} />
    </li>
  ));

  return (
    <ul>
      {repoElements}
    </ul>
  );
};

Repos.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    html_url: PropTypes.string.isRequired
  })).isRequired
};

export default Repos;
