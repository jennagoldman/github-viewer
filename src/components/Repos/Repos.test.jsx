import React from 'react';
import { shallow } from 'enzyme';
import Repos from './Repos.jsx';

describe('Repos', () => {
  it('matches a snapshot', () => {

    const repos = [
      { id: 1,
        name: 'rock-paper-scissors',
        html_url: 'https://github.com/jennagoldman/rock-paper-scissors' }
    ];

    const wrapper = shallow(<Repos repos={repos} />);
    expect(wrapper).toMatchSnapshot();
  });
});
