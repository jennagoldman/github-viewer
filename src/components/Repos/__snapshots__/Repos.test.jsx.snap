// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos matches a snapshot 1`] = `
<Fragment>
  <p>
    Repos:
  </p>
  <ul>
    <li
      key="1"
    >
      <Repo
        html_url="https://github.com/jennagoldman/rock-paper-scissors"
        id={1}
        name="rock-paper-scissors"
      />
    </li>
  </ul>
</Fragment>
`;
